temporalio/converter.py

```diff
@@ -149,7 +170,7 @@ async def encode(
             # converter may rely on the previous values
             payload = None
             for converter in self.converters.values():
-                payload = await converter.encode(value)
+                payload = converter.to_payload(value)
                 if payload is not None:
                     break
             if payload is None:
```

Comment:
In TS we decided to not throw from CompositePayloadConverter and instead throw in our helper functions.
This way you can use something other than CPC as your main converter and if you return `None` from `to_payload` we'll throw.

Reply:
I see, okay

Reply:
@lorensr this is pretty nice, WDYT? Should we port this to TS?

Reply:
Yeah, sounds like the middle ground we didn't consider between my returns-payload-or-throws and your don't-use-throwing-for-flow-control with `CompositePayloadConverter` 😄

The downside is that you couldn't use the same converter both as a standalone PayloadConverter and an  EncodingPayloadConverter with CompositePayloadConverter, but I'm fine with that. User can wrap the EncodingPayloadConverter with a throws-if-undefined converter.